homeassistant/components/withings/common.py

```diff
@@ -1030,6 +1030,21 @@ async def async_get_data_manager(
     if const.DATA_MANAGER not in config_entry_data:
         profile = config_entry.data.get(const.PROFILE)
 
+        if CONF_CLOUDHOOK_URL not in config_entry.data:
+            try:
+                webhook_url = await cloud.async_create_cloudhook(
+                    hass, config_entry.data[CONF_WEBHOOK_ID]
+                )
+            except cloud.CloudNotConnected:
+                webhook_url = webhook.async_generate_url(
+                    hass, config_entry.data[CONF_WEBHOOK_ID]
+                )
+            else:
+                data = {**config_entry.data, CONF_CLOUDHOOK_URL: webhook_url}
+                hass.config_entries.async_update_entry(config_entry, data=data)
+        else:
+            webhook_url = config_entry_data[CONF_CLOUDHOOK_URL]
```

Comment:
I got a KeyError when trying to access `config_entry_data`, but it seems to me as though this whole if block is testing whether the URL is safe to access in `config_entry.data` and that it should be this instead:

```suggestion
            webhook_url = config_entry.data[CONF_CLOUDHOOK_URL]
```

Reply:
In addition to this change, I found a number of steps were necessary to get this integration working:
https://github.com/home-assistant/core/issues/72614#issuecomment-1246218084

Importantly, subscriptions to the webhooks are always disabled AFAICT, and I needed to change the code for them to work at all:
https://github.com/home-assistant/core/issues/68760#issuecomment-1246206834